### DISCUSSION TITLE
why does this conditional not work?

### DISCUSSION THREAD

**Discussion by ryanpodonnell1** [redpanda-data/connect#2481] [General]
```yaml
      - label: extract_json_logs
        mutation: |
          #!blobl
          let msg = this.message.parse_json().or({})
          if $msg.exists("server_concurrent_connections") {
            root.haproxy = $msg
          } else {
            root = deleted()
          }
```

error: `failed to init processor 'extract_json_logs': failed to parse bloblang mapping '': expected =, got: $msg`

it seems to want a mapping as a result of the conditional

```
root = if .... {
}
```

following [this example](https://www.benthos.dev/docs/guides/bloblang/walkthrough/#if-statement) what i have _should_ work where I map inside the conditional

```
root = this
if this.pet.is_cute {
  root.pet.treats = this.pet.treats + 10
} else {
  root.pet.treats = deleted()
}
```

**Comment by mihaitodor** [comment]
Hey @ryanpodonnell1 👋 Are you using the latest version <https://github.com/benthosdev/benthos/releases/tag/v4.26.0>?

> Bloblang now supports root-level `if` statements.

**Reply by ryanpodonnell1** [reply]
ok thanks, let me bump (we're at 4.25) and give it a try!

**Reply by ryanpodonnell1** [reply]
that was it! thank you for the quick turnaround (as well as some other threads I've opened)

**Comment by ryanpodonnell1** [comment]
bumped to 4.26 and works now
